crates/matrix-sdk-base/src/sliding_sync.rs

```diff
@@ -329,7 +354,7 @@ impl BaseClient {
             .await?;
         }
 
-        let room_account_data = if let Some(events) = account_data.rooms.get(room_id) {
+        let room_account_data = if let Some(events) = &rooms_account_data.remove(room_id) {
             self.handle_room_account_data(room_id, events, changes).await;
             Some(events.to_vec())
```

Comment:
Taking the reference early forces you to reclone the same vector with the same content later. Could we do something like this instead:

```suggestion
        let room_account_data = if let Some(events) = rooms_account_data.remove(room_id) {
            self.handle_room_account_data(room_id, &events, changes).await;
            Some(events)
```